core/arch/arm/plat-hikey/conf.mk

```diff
@@ -24,6 +24,8 @@ CFG_NUM_THREADS ?= 8
 CFG_CRYPTO_WITH_CE ?= y
 CFG_WITH_STACK_CANARIES ?= y
 
+$(call force,CFG_DT,y)
```

Comment:
You definitely don't want to force this right now, otherwise people with an older ARM-TF (that does not pass a DT blob in r2) won't be able to boot.
So I think atm this should be a "documentation" patch, i.e., set `CFG_DT ?= n` and add a comment stating that HiKey _can_ read a DT blob if available, stating the benefits (i.e., get a unique the device ID form the eMMC CID), and that one needs a proper ARM-TF for this (give a Git URL if possible).
Later, when the "official" ARM-TF for HiKey -- be it linaro-swg or higher up in the upstream hierarchy -- we can default to ` ?= y`. And if some day we depend on DT features unconditionally, then we'll switch to `force`.